models/repo.go

```diff
@@ -654,7 +654,17 @@ func MigrateRepository(u *User, opts MigrateRepoOptions) (*Repository, error) {
 		return repo, UpdateRepository(repo, false)
 	}
 
-	if err = createUpdateHook(repoPath); err != nil {
+	repo, err = FinishMigrateRepository(repo, repoPath)
+	if err != nil {
+		return repo, err
+	}
+
+	return repo, UpdateRepository(repo, false)
+}
+
+// Finish migrating repository with things that don't need to be done for mirrors.
+func FinishMigrateRepository(repo *Repository, repoPath string) (*Repository, error) {
```

Comment:
I moved the existing code into a separate method and did not change anything (except for adding an additional return). The code I moved was not executed when creating a mirror but when creating a regular repo.

I thought this would be better than writing code that does essentially the same as the existing code.